src/coprocessor/dag/dag.rs

```diff
@@ -48,7 +48,18 @@ impl<S: Snapshot + 'static> DAGContext<S> {
         req_ctx: ReqContext,
         batch_row_limit: usize,
     ) -> Result<Self> {
-        let mut eval_cfg = box_try!(EvalConfig::new(req.get_time_zone_offset(), req.get_flags(),));
+        let mut eval_cfg = box_try!(EvalConfig::new(req.get_flags()));
+
+        // We respect time zone name first, then offset.
+        if req.has_time_zone_name() {
+            box_try!(eval_cfg.set_time_zone_by_name(req.get_time_zone_name()));
+        } else if req.has_time_zone_offset() {
+            box_try!(eval_cfg.set_time_zone_by_offset(req.get_time_zone_offset()));
+        } else {
+            // This should not be reachable. However we will not panic here in case
+            // of compatibility issues.
```

Comment:
Maybe it would be wise to raise a warning?

Reply:
This might be a warning that we could do nothing 🤣